src/components/NewAppModal.js

```diff
@@ -0,0 +1,168 @@
+import _ from 'lodash/fp'
+import { Fragment, useState } from 'react'
+import { div, h, li, span, ul } from 'react-hyperscript-helpers'
+import { ButtonPrimary, Link, spinnerOverlay } from 'src/components/common'
+import { icon } from 'src/components/icons'
+import { withModalDrawer } from 'src/components/ModalDrawer'
+import TitleBar from 'src/components/TitleBar'
+import { machineTypes } from 'src/data/machines'
+import { Ajax } from 'src/libs/ajax'
+import { currentApp } from 'src/libs/cluster-utils'
+import colors from 'src/libs/colors'
+import { withErrorReporting } from 'src/libs/error'
+import * as Style from 'src/libs/style'
+import * as Utils from 'src/libs/utils'
+
+
+const styles = {
+  whiteBoxContainer: { padding: '1rem', borderRadius: 3, backgroundColor: 'white' },
+  drawerContent: { display: 'flex', flexDirection: 'column', flex: '0 1 0%', padding: '0.5rem 1.5rem' },
+  headerText: { fontSize: 16, fontWeight: 600 }
+}
+
+// TODO: Once we have more appTypes, abstract text/component to support other apps
+export const NewAppModal = _.flow(
+  Utils.withDisplayName('NewAppModal'),
+  withModalDrawer({ width: 675 })
+)(({ onDismiss, onSuccess, namespace, apps, bucketName, workspaceName }) => {
+  const [viewMode, setViewMode] = useState(undefined)
+  const [loading, setLoading] = useState(false)
+
+
+  const app = currentApp(apps)
+
+  const createApp = _.flow(
+    Utils.withBusyState(() => setLoading(true)),
```

Comment:
`withBusyState` passes the busy value as an argument. If we hardcode `true` here, this won't clear the spinner if the call fails. Instead, because `setLoading` is already a function of one argument that does what we want, we can do:
```suggestion
    Utils.withBusyState(setLoading),
```